dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/CachingProviderService.java

```diff
@@ -81,19 +83,23 @@ public void activate(Map<String, Object> configProps) throws Exception {
          * TODO???? No doPriv due to limitations in OSGi and security manager. If
          * running with SecurityManager, permissions will need to be granted explicitly.
          */
-        CheckpointPhase.onRestore(0, () -> {
-            try {
-                ClassLoader classloader = getUnifiedClassLoader();
-                if (cachingProviderClass != null && !cachingProviderClass.trim().isEmpty()) {
-                    cachingProvider = Caching.getCachingProvider(cachingProviderClass, classloader);
-                } else {
-                    cachingProvider = Caching.getCachingProvider(classloader);
+        CheckpointPhase.onRestore(0, new OnRestore() {
```

Comment:
To make sure I understand, so all lambda's that have a catch will not have any FFDC processing?  I am curious how you found this issue?

Reply:
Thanks for your response! @jhanders34  actually found the root cause of the problem. In our case we were seeing consistent failures in `JCacheDynamicUpdateTest.dynamicUpdate`

Reply:
Thanks, that was a FULL test so explains why the PR builds allowed a merge.  Now I need to go review all the calls to onRestore that may have a similar issue.

Reply:
I'm not sure trace is a big deal, unless you need to see trace into the entry of the lambda (`call` method here).  Some would argue the body of the lambda should be trivial and not need entry/exit tracing.  For example, maybe the body of this lamda should have been extracted into a method that the onRestore lambda would simply reference/call.